OSOceanAcoustics/echopype
# Allow passing in `cal_params` for calibration

## Overview

Changes in this PR:
- addresses #887: allow feeding in data array with coordinates `('channel')` that encodes frequency-independent calibration parameter
- addresses #458: allow feeding in data array with coordinates `('cal_channel_id', 'cal_frequency')` that encodes frequency-dependent calibration parameter


## Details

### Immediately related to calibration code
- add new cal param ingestion functions in `calibration.cal_param.py`
  - `_get_interp_da`
  - `sanitize_user_cal_dict`
  - `param2da`
- update `get_cal_params_EK` and `get_cal_params_AZFP` to be the entry point for cal param intake
- in `CalibrateEK80` now all cal params come from `self.cal_params`
- squeeze and drop the `beam` dimension earlier in `CalibrateAZFP._cal_power_samples`

### Tests
- add tests for the new functions above
- add tests for `get_cal_params_EK` and `get_cal_params_AZFP`
- add test for `get_vend_cal_params_power`
- add integration tests to ensure the entire chain of intake is correct
